### PR TITLE
Fix validation errors on books page

### LINKED ISSUE #4440: Fix HTML validator issues on books page
Incorrect HTML can lead to rendering issues and bugs.

### Evidence / Screenshot (if possible)
The validator tool flags errors on https://validator.w3.org/nu/?doc=https%3A%2F%2Fopenlibrary.org%2Fbooks%2FOL28775995M%2FDer_Todeskanal/Der_Todeskanal

### Steps to Reproduce

1. Run `docker-compose exec web bash`
2. Run `python ./scripts/copydocs.py "/books/OL28775995M"
3. Use the validator tools direct input feature to flag errors
4. fix errors in the associated templates
5. submit a pull request addressing the issues.

### Stakeholders
@jdlrobson

### PR DESCRIPTION
**Pull request by Yashs911**
Closes #4440 and #3881

Fix validation errors on books page
### Screenshot

image
only duplicate id error is remaining that can be fixed on the homepage issue
### Stakeholders

@jdlrobson

### CODE REVIEW

**Comment by Yashs911** [comment]
Issue: #4447 should solve the duplicate ID errors

File: openlibrary/plugins/openlibrary/js/carousels_partials.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ export function initCarouselsPartials() {
             url: '/partials',
             type: 'GET',
             data: {
-                workid: $('.RelatedWorksCarousel').attr('workId'),
+                workid: $('.RelatedWorksCarousel').attr('data-workId'),
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/carousels_partials.js:12]: While this works the same, it's better in jquery to use `.data('workId').` as it has no DOM lookup and can be read from cache.

File: openlibrary/templates/lib/nav_head.html [hunk 2]
```diff
@@ -3,7 +3,7 @@
 <header id="header-bar" class="header-bar">
   <div class="hamburger-component">
 
-    <label role="button" for="toggle-hamburger" class="hamburger-button">
+    <label for="toggle-hamburger" class="hamburger-button">
```
Comment (jdlrobson) [inline openlibrary/templates/lib/nav_head.html:6]: The role is there for accessibility. Am curious what the error was here. I think the role needs to be on the input[typecheckbox] on L10 instead.

**Review by jdlrobson — CHANGES_REQUESTED** [review]
2 tiny nitpicks! Sorry for the delay with this review!

File: openlibrary/templates/lib/nav_head.html (hunk 2 above)
Comment (Yashs911) [reply openlibrary/templates/lib/nav_head.html:6]:
```
Error: Attribute role not allowed on element label at this point.

From line 107, column 5; to line 107, column 73

nt">↩↩    <label role="button" for="toggle-hamburger" class="hamburger-button">↩     

Attributes for element label:
    Global attributes
    for — Associate the label with form control
```

Comment (jdlrobson) [reply openlibrary/templates/lib/nav_head.html:6]: Yep so looks like this was meant for the input[type=checkbox]. Nice catch!

**Comment by Yashs911** [comment]
@jdlrobson I have made the required changes.

**Comment by jdlrobson** [comment]
Looks great! The only remaining error is around the `read_ebook` id which is being fixed in #4460
Thanks for making this happen!

**Pull request merged**